src/synth/fluid_chan.c

```diff
@@ -758,3 +758,69 @@ void fluid_channel_set_override_gen_default(fluid_channel_t *chan, int gen, flui
     chan->override_gen_default[gen].flags = GEN_SET;
     chan->override_gen_default[gen].val = val;
 }
+
+/* Calculate portamento time in milliseconds considering the synthesizer's portamento time mode */
+unsigned int fluid_channel_portamentotime_with_mode(fluid_channel_t *chan, enum fluid_portamento_time_mode time_mode, int lsb_seen, int fromkey, int tokey)
+{
+    int msb = fluid_channel_get_cc(chan, PORTAMENTO_TIME_MSB);
+    int lsb = fluid_channel_get_cc(chan, PORTAMENTO_TIME_LSB);
+    int res;
+    fluid_real_t tmp;
+    static const int Max = 480*1000; /*ms*/
+    
+    /* Use 7-bit MSB initially, switch to 14-bit if LSB has been seen */
+    if(time_mode == FLUID_PORTAMENTO_TIME_MODE_AUTO)
+    {
+        if(lsb_seen)
+        {
+            time_mode = FLUID_PORTAMENTO_TIME_MODE_LINEAR;
+        }
+        else
+        {
+            time_mode = FLUID_PORTAMENTO_TIME_MODE_XG_GS;
+        }
+    }
+
+    switch(time_mode)
+    {
+        case FLUID_PORTAMENTO_TIME_MODE_XG_GS:
+            // Produce a curve similar to:
+            /*
+                CC 5 value  Portamento time
+                ----------  ---------------
+                    0            0.000 s
+                    1            0.006 s
+                    2            0.023 s
+                    4            0.050 s
+                    8            0.110 s
+                    16           0.250 s
+                    32           0.500 s
+                    64           2.060 s
+                    80           4.200 s
+                    96           8.400 s
+                    112         19.500 s
+                    116         26.700 s
+                    120         40.000 s
+                    124         80.000 s
+                    127        480.000 s
+            */
+            // Tests were performed by John Novak
+            // https://github.com/dosbox-staging/dosbox-staging/pull/2705
+            tmp = fluid_concave(msb);
+            res = (Max/2 * 2.5) * tmp * fluid_concave(126 * tmp) + 200 * fluid_convex(msb);
+            res = res < Max ? res : Max;
+            // Apply a similar scaling hack as SpessaSynth to fix Descent Game08, it's unclear why exactly
+            // https://github.com/spessasus/spessasynth_core/blob/0b2d44f48065d3d6bbca24a1d40223b1255dab00/src/synthesizer/audio_engine/engine_methods/portamento_time.ts#L84-L86
```

Comment:
```suggestion
            // https://github.com/spessasus/spessasynth_core/blob/5a8730a80f8c0b74733ec193a968b36e2a0c0aee/src/synthesizer/audio_engine/engine_methods/portamento_time.ts#L84-87
```
This link is outdated now. I would suggest updating it to the 36 semitone version, to not confuse the readers (i.e. why is it 36 there and 30 in spessasynth?)